packages/analytics-browser/src/browser-client.ts

```diff
@@ -83,10 +84,17 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
 
     // Step 3: Set session ID
     // Priority 1: `options.sessionId`
-    // Priority 2: last known sessionId from user identity storage
+    // Priority 2: sessionId from url
+    // Priority 3: last known sessionId from user identity storage
     // Default: `Date.now()`
     // Session ID is handled differently than device ID and user ID due to session events
-    this.setSessionId(options.sessionId ?? this.config.sessionId ?? Date.now());
+    const queryParams = getQueryParams();
+    this.setSessionId(
+      options.sessionId ??
+        (queryParams.ampSessionId ? Number(queryParams.ampSessionId) : undefined) ??
```

Comment:
Not 100% that this convert Number logic works as expected.

```
let ampSessionId = "hi"
let sessionId = (ampSessionId ? Number(ampSessionId) : undefined) ?? 42
console.log(sessionId) // 42
```

Reply:
Updated at https://github.com/amplitude/Amplitude-TypeScript/pull/753/commits/c5b79601bf525d532afd157572c28617209db9ef